### ISSUE TITLE
Integrate labels into rewards.rchain.coop 

### ISSUE THREAD

**Issue by ICA3DaR5** [rchain/bounties#673]
_opened on behalf of @dckc_

### Benefit to RChain Coop and Community

[Guides](https://github.com/rchain/bounties/wiki/Bounty-Task-Guides) could easily review budgets on their issues.

### Proposed Budget

**$300** based on an estimate of 2 story points ($125 ea per #273) plus some training time.

### Measure of Completion

 - [x] rewards.rchain.coop sync integrates labels
 - [x] [guides][] are trained to use them

[guides]: https://github.com/rchain/bounties/wiki/Bounty-Task-Guides

----
#### Legal
*Task Submitter shall not submit Tasks that will involve RHOC being transacted in any manner that (i) jeopardizes RHOC’s status as a software access token or other relevant and applicable description of the RHOC as an “asset”—not a security— or (2) violates, in any manner, applicable U.S. Securities laws.*

**Comment by jimscarver** [comment]
I started making this table for review of labeled issues https://rhobot.glitch.me/
As is the table can be copy pasted into a google sheet for cooperation amoung the reviewers of the label.
Here is the query I use

  repository(owner: "rchain", name: "bounties") {
    issues(first: 100, labels: ["Governance"], states: [OPEN], orderBy: {field: UPDATED_AT, direction: DESC}) {
      nodes {
        number
        title
        author {login}
        updatedAt
        labels(first: 10) {
          nodes {
            name
          }
        }
      }
    }
  }
}

**Comment by jimscarver** [comment]
@dckc  We can extend https://rhobot.glitch.me/ to provide information from the mysql bounty database given credentials. Links into the bounty system  for voting  might also be good.  Do you think this is a worthwhile activity or are there other plans?

**Comment by dckc** [comment]
It's probably worthwhile, especially if it happens before I find time for my idea: I intend to enhance the issue download stuff in rewards.rchain.coop to grab labels.

No credentials are needed to access any of the data in rewards.rchain.coop; it's all public. I think you can GET any data you like in CSV format... maybe JSON or XML too.

**Comment by lapin7** [comment]
Yeah, that's very powerful. Is all data from August 2017 onwards in the App?

**Comment by jimscarver** [comment]
@dckc  We will want to access the bounty data real time but can use a snapshot for development.
Can you suggest how we might make a link to vote on an issue?

**Comment by dckc** [comment]
> Can you suggest how we might make a link to vote on an issue?

Err... navigate to it and bookmark the URL in your browser as usual. Is there some trick to this that I'm missing?

e.g. [all votes on 668 in CSV][3], [a vote related to 668 in HTML][1], [vote exported in XML][2]

> Is all data from August 2017 onwards in the App?

Sorting the rewards by date shows that it's not, right? Again, is there a trick to this that I'm missing?

It's not clear to me how much you tried to find the answer yourself before putting the ball in my court. [How to Ask Questions The Smart Way](http://www.catb.org/esr/faqs/smart-questions.html#before) is largely common sense, but evidently it bears repeating:

> When you ask your question, display the fact that you have [tried various ways to find the answer] first; this will help establish that you're not being a lazy sponge and wasting people's time.

[1]: https://rewards.rchain.coop/index.php?-action=view_related_record&-related-record-id=issue%2FBudgetVotes%3Fnum%3D668%26BudgetVotes%253A%253Apay_period%3D2018-05-01%26BudgetVotes%253A%253Aissue_num%3D668%26BudgetVotes%253A%253Avoter%3DOjimadu&-table=issue&num=%3D668&-ui-root=main-content&-sort=num+desc&-cursor=0&-skip=0&-limit=30&-mode=list
[2]: https://rewards.rchain.coop/index.php?-action=export_xml&-table=issue&num=%3D668&-ui-root=main-content&--single-record-only=1
[3]: https://rewards.rchain.coop/index.php?-table=budget_vote&-limit=999999&issue_num=668&amount=&vote_time=&-action=export_csv&-cursor=0&-skip=0&-mode=list

**Comment by lapin7** [comment]
Here's a spreadsheet to play with.
https://docs.google.com/spreadsheets/d/1UfRgtSaTcZvuVW7GOonivA3Qgs2xVgT-0SVwpuDxHzM/edit#gid=592953635

It has all data of Budgets and Bounties from August 2017 to April 2018

It's the base for a nice financial report.

**Comment by dckc** [comment]
data dump: http://rewards.rchain.coop/db-bak/201804.sql.gz

**Comment by dckc** [comment]
code to sync labels is committed and deployed:
 - https://github.com/dckc/rchain-dbr d3af715

The Right Thing would have been a new `issue_label` table to relate labels to issues; but I took a short-cut by just adding labels to the `issue` table.

Still to do: train guides.

**Comment by aviationhacker** [comment]
Just an update from Jim & myself, we've cloned the open issues table/site we built (https://rchain-bounty.glitch.me), and are now updating it with a GraphQL element and a MySQL element, looking to connect the two to basically add more functions in the future. The original site (https://rhobot.glitch.me) has been left as is, as we understand some people are using it to help them with other tasks.

At this point, if anyone is pretty good with GraphQL and feels like coaching us a little, we could definitely use the help!

**Comment by jimscarver** [comment]
Anyone who knows some javascript or is willing to learn GraphQL and not too afraid of typing commands should join our collaborative learning work studies Thursday 2pm EST, Mon, Tue, Thu 4pm EST in https://zoom.us/j/6853551826

I expect the rhobot project will fork from this issue once the label review app is complete.

**Comment by dckc** [comment]
I looked into glitch a little... pretty cool!

I know javascript and GraphQL, but I'm not available at those times. I'm available Saturdays at 9am (#403).

**Comment by dckc** [comment]
@lapin7 thanks for the suggestion. How does it look?

 - `088f643` include labels in reward, budget views

**Comment by dckc** [comment]
I have talked with several guides about this: @lapin7 , @TrenchFloat , @pmoorman , @jimscarver .

Optimistically, I'm going to say that's a critical mass and they'll train the rest.

**Comment by jimscarver** [comment]
Prototyping support for the review process is continuing @dckc Is there someplace else to report that oe should this issue be reopened? 
We added creation date and column sorting per member request to http://rhobot.glitch.me
Cookies to remember choices, more cross links to rewards and cosmetic improvement are in the pipe.

**Comment by dckc** [comment]
Good question. Stand by for a substantive reply. If you don't see one in a
few days, please remind me.

**Comment by dckc** [comment]
I wonder how to integrate rhobot.glitch.me into rewards.rchain.coop

But I suppose it's good to keep the discussion here for a bit...

**Comment by jimscarver** [comment]
The answer is you don't  integrate rhobot.glitch.me into rewards.rchain.coop. rhobot is a user driven interface independent of the rewards.rchain.coop interface. It intergrates via APIs. We can integrate our own fron ends using mysql but it would be nice if APIs for each function were available, preferably with GraphQL

Todaty Allanc, JIm, Gary and Aadi Reordered columns, added ALL to label dropdown
Added get issue by number and css in https://rhobot.glitch.me/

**Comment by dckc** [comment]
> The answer is you don't integrate rhobot.glitch.me into rewards.rchain.coop.

Ok... then since this issue is about rewards.rchain.coop, that work is out of scope of this issue. Let's use #609 to track it.

**Comment by dckc** [comment]
@jimscarver @aviationhacker given the scope clarification above, your budget votes on this issue look a bit high. Please review.

**Comment by aviationhacker** [comment]
I've just had a look at my budget for this issue, but I can't see one that I've placed? Was it removed? If so, it saves me changing it :)

**Comment by dckc** [comment]
I guess I'm confusing you with @entropee .

**Comment by pmoorman** [comment]
@dckc not sure if this is the right place (but it's better than nothing):

small feature request for the rewards app: **Could we add the label "discussion" to the labels that would exclude an issue from the Overdue list?** 

Discussion issues often don't require a vote. #775 and #831 are examples of that nature (and sit in our marketing overdue list)

**Comment by dckc** [comment]
good idea

**Comment by dckc** [comment]
I made the Discussion tweak 2018-09-26 in 121343ca
